exercise_2/RecordsHandler.hpp

```diff
@@ -0,0 +1,45 @@
+//
+// Created by denys on 10.07.19.
+//
+
+#pragma once
+
+
+#include <map>
+#include <memory>
+#include "Course.hpp"
+#include "Exam.hpp"
+#include "Student.hpp"
+#include "Teacher.hpp"
+
+
+class RecordsHandler {
+private:
+    template <typename T>
+    static std::shared_ptr<IRecord> _constructor(std::istringstream &istringstream) {
+        return std::make_shared<T>(istringstream);
+    }
+    using _constructor_type = std::shared_ptr<IRecord> (*)(std::istringstream &);
+    std::map<std::string, _constructor_type>  CONSTRUCTORS;
```

Comment:
I stronly believe that shall be of form:
`std::map<std::string, IRecord*> `
or
`std::map<std::string, IRecord&> `

Reply:
Effectively, the _constructor method won't be needed.
Effectively, you won't need CONSTRUCTORS container
Effectively, you will be able to spawn proper record ad hod (by make_shared or in any other convenient way).

Reply:
My records collections is defined in 24 line.

I was trying to implement some Factory pattern features. So in my case, if I need one more record reader I just need to create class that inherits from IRecord and add record type in the RecordsHandler::RecordsHandler() constructor.